### PR TITLE
label room ids

### PR DESCRIPTION
**Pull request by zelosos**
- gave each value a label
- replaced hex values by label
- commented unknown labels if unsure

### CODE REVIEW

File: src/code/bank0.asm [hunk 1]
```diff
@@ -6604,24 +6604,24 @@ label_3500::
...
+    cp   UNKNOWN_ROOM_0C                          ; $3517: $FE $0C
     ret  z                                        ; $3519: $C8
-    cp   $1B            ; The drainable water above Angler's Tunnel ; $351A: $FE $1B
+    cp   UNKNOWN_ROOM_1B                          ; $351A: $FE $1B
     ret  z                                        ; $351C: $C8
 
 label_351D::
     ldh  a, [hMapRoom]                            ; $351D: $F0 $F6
-    cp   $80            ; Mysterious Forest enterance from Mabe (?) ; $351F: $FE $80
+    cp   IS_INDOOR                                ; $351F: $FE $80
```
Comment (kemenaran) [inline src/code/bank0.asm:6624]: Map rooms go from $00 to $FF on all maps, overworld and indoor. Not sure if this `IS_INDOOR` is correct then. It's more probably identifying the first part of the overworld rooms?

**Review by kemenaran — CHANGES_REQUESTED** [review]
Thanks for this work. IMHO clearly identify the room constants makes the code easier to read.

To move this PR forward, I would suggest:

1. **To use a consistent naming**, instead of having some constants prefixed with `OVERWORLD`, some with `INDOOR`, some with none at all, and some missing the `ROOM` suffix.
    For instance, how about using a `ROOM_<base-map>_<name>` naming scheme? Yielding something like `ROOM_OVERWORLD_EGG`, `ROOM_INDOORS_A_KANALET_UNDERGROUND`, etc. This will also allow to neatly group constants by base map, grouping all `ROOM_OVERWORLD_*`, `ROOM_INDOORS_A_*` and `ROOM_INDOORS_B_*` together.
2. **Remove unused variables.** I don't think we need to declare all `OVERWORLD_F4` rooms, especially if they are not used. Only declaring the used room constants would already go a long way.

I think this could allow to merge this PR neatly.

Also, as a bonus, it would be nice to properly name all the room constants used in the code (like `UNKNOWN_ROOM_0C` -> `ROOM_OVERWORLD_TARAMANCH_BRIDGE`). [XKeeper's map viewer](https://xkeeper.net/hacking/linksawakening/test.php) could be used to identify the room from it's index. But I'm fine with merging this PR without it, and rename the constants to room names later.

File: src/code/bank0.asm [hunk 2]
```diff
@@ -5157,14 +5157,14 @@ LoadTileset9::
...
+    cp   INDOOR_MANBO_THE_SUN_FISH_ROOM           ; $2EA8: $FE $FD
     jr   z, .label_2ED3                           ; $2EAA: $28 $27
-    cp   $B1                                      ; $2EAC: $FE $B1
+    cp   FISHING_MINIGAME_ROOM                    ; $2EAC: $FE $B1
     jr   z, .label_2ED3                           ; $2EAE: $28 $23
 
 .label_2EB0
     ld   a, [wIsBowWowFollowingLink]              ; $2EB0: $FA $56 $DB
-    cp   $01                                      ; $2EB3: $FE $01
+    cp   TRUE                                     ; $2EB3: $FE $01
```
Comment (daid) [inline src/code/bank0.asm:5167]: This is "wrong". wIsBowWowFollowingLink can have more values then 00 and 01. If I remember right it is 80 when bowwow is kidnapped.

File: src/code/bank0.asm [hunk 3]
```diff
@@ -3644,7 +3644,7 @@ ENDC
     ldh  a, [hMapRoom]                            ; $2049: $F0 $F6
     ld   e, a                                     ; $204B: $5F
     ld   d, $00                                   ; $204C: $16 $00
-    ld   a, $14                                   ; $204E: $3E $14
+    ld   a, BANK(SignpostDialogTable)             ; $204E: $3E $14
```
[the remark this answers is not in the document]
Comment (zelosos) [reply src/code/bank0.asm:3647]: Thanks :)

File: src/code/bank0.asm (hunk 2 above)
Comment (zelosos) [reply src/code/bank0.asm:5167]: ok. Then I will revert that.

**Comment by zelosos** [comment]
What does the A in ROOM_INDOORS_A_* stand for?
Since I never played Zelda through and also played the german version, I would not set the names for the rooms.
I do not know the right names.

**Comment by daid** [comment]
I get the feeling you lack some basic knowledge about how the maps are "put together".

Simply put there are 4 maps:
- Overworld
- Indoor A
- Indoor B
- Color dungeon
Except for the color dungeon, these maps are 16x16 rooms each. And can be seen here:
http://artemis251.fobby.net/zelda/
https://xkeeper.net/hacking/linksawakening/test.php
Dungeons are actually composed out of rooms from these indoor maps, and are just a "view" on top of these. (Well, there is some more complexity with tilesets and stuff, but I'm ignoring that for now)

File: src/code/bank0.asm [hunk 4]
```diff
@@ -5901,7 +5901,7 @@ LoadRoom::
 .loadBankForOverworldRooms
     ; … and the overworld room index is >= $80…
     ldh  a, [hMapRoom]                            ; $322F: $F0 $F6
-    cp   $80                                      ; $3231: $FE $80
+    cp   IS_INDOOR                                ; $3231: $FE $80
```
Comment (daid) [inline src/code/bank0.asm:5904]: This checks for lower upper half of the overworld, as data for those is split 50/50 in 2 banks.

File: src/code/bank1.asm [hunk 5]
```diff
@@ -2005,7 +2005,7 @@ CreateFollowingNpcEntity::
     and  a                                        ; $6022: $A7
     jr   nz, .ghostEnd                            ; $6023: $20 $5A
     ldh  a, [hMapRoom]                            ; $6025: $F0 $F6
-    cp   $40                                      ; $6027: $FE $40
+    cp   IS_UNKWON_ROOM_CATEGORY                  ; $6027: $FE $40
```
Comment (daid) [inline src/code/bank1.asm:2008]: This doesn't check for a specific room, but causes the ghost to follow you when you leave the upper 25% of the overworld map after finishing dungeon 4.

File: src/code/bank20.asm [hunk 6]
```diff
@@ -5261,84 +5224,78 @@ LoadRoomObjectsAttributes::
...
     jr   .jr_020_6E14                             ; $6DF4: $18 $1E
 
 .jr_020_6DF6
-    cp   $1B                                      ; $6DF6: $FE $1B
+    cp   UNKNOWN_ROOM_1B                          ; $6DF6: $FE $1B
     jr   nz, .jr_020_6E06                         ; $6DF8: $20 $0C
-
-    ld   a, [wOverworldRoomStatus + $2B]                               ; $6DFA: $FA $2B $D8
-    and  $10                                      ; $6DFD: $E6 $10
+    ; is this here a copy and past bug? should it be $1B ?
```
Comment (daid) [inline src/code/bank20.asm:5269]: Nope, this changes the upper waterfall above D4 when D4 is opened.

**Comment by zelosos** [comment]
Sorry I have not the time to complete the pull request.
Use / improve it or throw it away. It is your choice.

**Comment by kemenaran** [comment]
Sure, I'll try to have a shot at updating this PR and put it in a mergeable state.

Thanks for your work!

**Comment by kemenaran** [comment]
I updated this PR, with :

- Proper prefixing of room constants
- Rooms constants are now sorted by Overworld/IndoorA/IndooB
- Lot of cleanup

There are still many unlabeled room constants, but they are clearly indicated as such, and can be labeled later.

@daid what do you think?

**Review by kemenaran — APPROVED** [review]

**Pull request merged**